crucible-debug/README.md

```diff
@@ -0,0 +1,33 @@
+# crucible-debug
+
+`crucible-debug` is a library for interactive debugging of Crucible programs.
+It is structured as an [`ExecutionFeature`][exec-feat], meaning that it can be used in situ inside of any program that uses Crucible.
+
+[exec-feat]: https://hackage.haskell.org/package/crucible-0.7.1/docs/Lang-Crucible-Simulator.html#t:ExecutionFeature
+
+Users interact with the debugger via a simple command language.
+This language features a number of familiar commands such as setting and removing breakpoints (`break`/`delete`), printing stack traces (`backtrace`), stepping through symbolic execution (`step`), and loading debugger commands from files (`source`).
+It additionally provides more Crucible-specific commands such as inspecting program CFGs (`cfg`) and printing the current path condition (`path-condition`).
+All available commands can be listed using the interactive documentation system (`help`).
+
+In addition to the library, this package provides an executable.
+The executable begins the debugger inside of a "minimal" Crucible program that immediately returns the unit value `()`.
+This is not very interesting on its own, but the `load` command can load Crucible S-expression programs, and the `call` command can call functions from loaded programs.
+
+The command language can be extended by downstream packages.
+Such extensions can be used to add commands that are specific to a particular Crucible syntax extension (i.e., source language).
+For example, the `crucible-llvm-debug` package provides the `memory` command to print the current LLVM memory.
+
+The test suite is described in its Haddocks.
```

Comment:
It might be worth a brief comparison with [`surveyor`](https://github.com/GaloisInc/surveyor) (or any other prior art that you can think of) here.